### DISCUSSION TITLE
Does vcpkg community has the plan to provide a vcpkg cli? If not why?

### DISCUSSION THREAD

**Discussion by aiyolo** [microsoft/vcpkg#42048] [Ideas]
A vcpkg cli will be much more convienient for developers.

**Comment by dg0yt** [comment]
What is a vcpkg cli?
Did I miss a vcpkg GUI or lib?

**Reply by aiyolo** [reply]
I mean a command line executable, like cargo.

**Reply by dg0yt** [reply]
What is missing from the `vcpkg` command line executable?

**Comment by dg0yt** [comment]
https://learn.microsoft.com/en-us/vcpkg/concepts/classic-mode
https://learn.microsoft.com/en-us/vcpkg/commands/install
https://learn.microsoft.com/en-us/vcpkg/commands/common-options

**Reply by dg0yt** [reply]
But maybe not easily discoverable on the Microsoft pages?

**Reply by aiyolo** [reply]
I'm sorry that I didn't explain my question clearly. 

Will vcpkg provide a installer in the future? instead  of cloning a git repository? if not, what's the concern?

**Reply by dg0yt** [reply]
Which benefit do you expect from an installer? `git` is how versioning works in vcpkg (manifest mode). The executable is the only binary artifact.

And IIUC, Visual Studio comes with a manifest-mode-only bundled vcpkg.

**Reply by aiyolo** [reply]
My concerns are:
1. The speed of cloning the vcpkg repo from github is very slow in my area for some network reasons.
2. Cloning vcpkg repo and then do some bootstrap action is no easier than simply using a installer.

**Reply by dg0yt** [reply]
> The speed of cloning the vcpkg repo from github is very slow in my area for some network reasons.

You don't have to clone the full history. 
You can create more clones from a single local checkout.
If you don't use versioning, you could even download a tarball/ZIP archive.

And vcpkg is building packages from source. Downloads will happen all the time.

> Cloning vcpkg repo and then do some bootstrap action is no easier than simply using a installer.

Hm, if typing two short commands isn't simple enough, then I wonder if using vcpkg will ever be simple enough.

**Reply by aiyolo** [reply]
Thanks god, I finaly found that scoop can help me do all the thing. 
with a single command
```
scoop install vcpkg
```
scoop wiil dowload vcpkg.zip and run bootstrap.bat and run integrate install.
https://github.com/ScoopInstaller/Main/blob/master/bucket/vcpkg.json

**Reply by dg0yt** [reply]
There is a long record of issues hitting users of scoop... 
https://github.com/microsoft/vcpkg/issues?q=is%3Aissue%20scoop
